utils.py

```diff
@@ -891,11 +891,22 @@ def check_auth(user, hutch):
 def check_special(ioc, hutch):
     lines = open(SPECIAL_FILE % hutch).readlines()
     lines = [l.strip() for l in lines]
-    for l in lines:
+    for l in lines: #Note: (josh) as long as the ioc name is in the special file entry we let it be cycled by any user 
         if l == ioc:
             return True
     return False
 
+# checks if ioc is marked as toggleable between defined versions of IOCs
+# schema will be ioc_name:permittedversion1,permittedversion2,etc
+def check_special_toggle(ioc, hutch, version):
+    lines = open(SPECIAL_FILE % hutch).readlines()
+    for l in lines:
+        permitted_versions = l.split(":").split(",")
```

Comment:
l.split(":") is a list with no split method.

Also, you need to test the first element, to check that it is the ioc in question.  (You don't want to allow *any* IOC to upgrade to one of the allowed gem versions.)